generate_specs.py

```diff
@@ -97,61 +152,61 @@ def generate_spec(
         elif other_func.end_line == end_line and other_func.end_col >= end_col:
             other_func.end_col += new_end_col - end_col
 
-    with Path(out_file).open(mode="w", encoding="utf-8") as f:
-        f.write(new_contents)
+    Path(out_file).write_text(new_contents)
 
 
 def recover_from_failure() -> None:
     """Implement recovery logic."""
     raise NotImplementedError("TODO: Implement me")
 
 
-def verify_one_function(func_name: str, llvm_analysis: LLVMAnalysis, out_file: Path) -> bool | None:
+def verify_one_function(
+    prompt_builder: PromptBuilder,
+    func_name: str,
+    parsec_result: ParsecResult,
+    processed_funcs: list[str],
+    out_file: Path,
+) -> bool | None:
     """Return the result of running CBMC on the specifications generated for a single function.
 
     Args:
+        prompt_builder (PromptBuilder): The prompt builder to construct prompts for the LLM.
         func_name (str): The name of the function to verify.
-        llvm_analysis (LLVMAnalysis): The top-level LLVM analysis.
-        out_file (Path): The path to the updated file where the function is defined (with specs).
+        parsec_result (ParsecResult): The result of running `parsec` on the initial code.
+        processed_funcs (list[str]): The names of previously-processed functions.
+        out_file (Path): The path to the updated file where the function is defined (with a spec).
 
     Returns:
         bool | None: True iff verification succeeds, False if it fails. None if the function is
```

Comment:
The code below now raises `RuntimeError` rather than returning `None` (which I think is better behavior).